### PR TITLE
Revert "Field `status.hostIPs` added for Pod (#101566)"

### PR DESCRIPTION
**Pull request by Argh4k**
This reverts commit 61b3c028ba618a939559c39befb546ae5e5fd0b9 from https://github.com/kubernetes/kubernetes/pull/101566

#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

xref https://github.com/kubernetes/kubernetes/issues/109146

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by dims** [comment]
cc @wzshiming as author of https://github.com/kubernetes/kubernetes/pull/101566 ( also @aojea )

**Comment by Argh4k** [comment]
From pull-kubernetes-e2e-gce-100-performance. 50% percentile:
```
    {
      "Name": "kube-apiserver-e2e-109151-95a39-master/kube-apiserver",
      "CPU": 0.585443117,
      "Mem": 1526702080
    },
```

**Comment by liggitt** [comment]
/retest

integration timeout

**Comment by liggitt** [comment]
> @liggitt
> 
> > the kubelet status reconciliation is not setting the field and thinks it needs to be reconciled
> 
> I'm out of touch with Kubelet - is this different than `generateAPIPodStatus()` ? Or is the error because conversion was setting the value whjen it shouldn't?

I suspect both were involved... I think this is what happens when the API server returns the field and the kubelet omits setting it in that function because the feature gate is off.

That said, that reveals a skew problem that would occur when the API server enabled the feature in the future and old kubelets were still not setting it because their version had the feature disabled.

**Comment by thockin** [comment]
So we should copy the old values in `convertStatusToAPIStatus` I guess.

**Comment by aojea** [comment]
/retest
```
Kubernetes e2e suite: [sig-storage] CSI mock volume CSI FSGroupPolicy [LinuxOnly] should modify fsGroup if fsGroupPolicy=File expand_more	6m59s
Kubernetes e2e suite: [sig-storage] In-tree Volumes [Driver: local][LocalVolumeType: dir-bindmounted] [Testpattern: Pre-provisioned PV (default fs)] subPath should be able to unmount after the subpath directory is deleted [LinuxOnly] expand_more	2m13s
Kubernetes e2e suite: [sig-node] Probing container should *not* be restarted by liveness probe because startup probe delays it expand_more	4m14s
Kubernetes e2e suite: [sig-cli] Kubectl client Simple pod should return command exit codes running a successful command expand_more	5m10s
Kubernetes e2e suite: [sig-apps] StatefulSet Basic StatefulSet functionality [StatefulSetBasic] should perform canary updates and phased rolling updates of template modifications [Conformance] expand_more	10m36s
Kubernetes e2e suite: [sig-apps] StatefulSet Basic StatefulSet functionality [StatefulSetBasic] should implement legacy replacement when the update strategy is OnDelete expand_more	10m36s
Kubernetes e2e suite: [sig-apps] Deployment iterative rollouts should eventually progress expand_more	5m50s
Kubernetes e2e suite: [sig-apps] StatefulSet AvailableReplicas should get updated accordingly when MinReadySeconds is enabled expand_more	10m18s
Kubernetes e2e suite: [sig-apps] StatefulSet Basic StatefulSet functionality [StatefulSetBasic] should list, patch and delete a collection of StatefulSets [Conformance] expand_more	10m24s
Kubernetes e2e suite: [sig-apps] Deployment should run the lifecycle of a Deployment [Conformance] expand_more	5m24s
Kubernetes e2e suite: [sig-apps] StatefulSet Basic StatefulSet functionality [StatefulSetBasic] should perform rolling updates and roll backs of template modifications [Conformance] expand_more	10m36s
Kubernetes e2e suite: [sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Dynamic PV (default fs)] volumes should store data expand_more	2m17s
Kubernetes e2e suite: [sig-storage] Volumes GlusterFS should be mountable expand_more	2m48s
Kubernetes e2e suite: [sig-apps] Deployment deployment should support proportional scaling [Conformance] expand_more	5m26s
Kubernetes e2e suite: [sig-apps] ReplicaSet Replace and Patch tests [Conformance] expand_more
```

and
```

k8s.io/kubernetes/test/integration/apiserver: TestPatchConflicts expand_more	5m5s
k8s.io/kubernetes/test/integration/apiserver/admissionwebhook: TestWebhookTimeoutWithoutWatchCache/webhooks_consume_client_timeout_available,_not_webhook_timeout expand_more	5m3s
k8s.io/kubernetes/test/integration/apiserver/admissionwebhook: TestWebhookTimeoutWithoutWatchCache expand_more	5m12s
k8s.io/kubernetes/test/integration/auth: TestPodSecurityWebhook expand_more	5m8s
k8s.io/kubernetes/test/integration/client: TestEventCompatibility expand_more	5m8s
k8s.io/kubernetes/test/integration/client: TestExecPluginRotationViaInformer expand_more	0s
k8s.io/kubernetes/test/integration/controlplane: TestGracefulShutdown expand_more
```

**Comment by liggitt** [comment]
flake rate on this PR is 89% :-(

**Comment by pacoxu** [comment]
https://testgrid.k8s.io/sig-release-master-informing#gce-master-scale-performance 
The failure is also waiting for this revert.

**Comment by aojea** [comment]
/test pull-kubernetes-integration
job timeout

**Comment by Argh4k** [comment]
Do we know what is causing this flakiness? Is this something normal or has it increased lately?

**Comment by wzshiming** [comment]
> > @liggitt
> > > the kubelet status reconciliation is not setting the field and thinks it needs to be reconciled
> > 
> > 
> > I'm out of touch with Kubelet - is this different than `generateAPIPodStatus()` ? Or is the error because conversion was setting the value whjen it shouldn't?
> 
> I suspect both were involved... I think this is what happens when the API server returns the field and the kubelet omits setting it in that function because the feature gate is off.
> 
> That said, even if we fixed the issue with the API server populating the field in alpha, that reveals a skew problem that would occur when the API server enabled the feature in the future and old kubelets were still not setting it because their version had the feature disabled.

https://github.com/kubernetes/kubernetes/pull/109151/files#diff-d8d6f2653252d254879b62c9a54c91fc9a493436281e3fa6ad538330ae16c816L273-L283

Is need to add Featuregate to Convert, Avoid fill fields when feature is disabled?

/lgtm

**Comment by Argh4k** [comment]
Rebased.

**Comment by liggitt** [comment]
openapi-spec needs regen

File: api/openapi-spec/swagger.json [hunk 1]
```diff
@@ -15254,89 +15234,6 @@
       },
       "type": "object"
     },
-    "io.k8s.api.storage.v1alpha1.CSIStorageCapacity": {
```
Comment (liggitt) [inline api/openapi-spec/swagger.json:15257]: this doesn't belong in this PR... just realized why this kicked in. beta.0 was just tagged and this resource is marked to stop being served in 1.24, and that automatically turns off at beta

I'll make a separate commit you can rebase on that just has this change

Comment (liggitt) [reply api/openapi-spec/swagger.json:15257]: https://github.com/kubernetes/kubernetes/pull/109195 is open that just regenerates openapi on master

**Comment by liggitt** [comment]
> OK, maybe I can insist on adding FeatureGate in convert at that time, maybe I can avoid this result.

conversion cannot use feature gates, it needs to be stateless

File: api/openapi-spec/swagger.json (hunk 1 above)
Comment (Argh4k) [reply api/openapi-spec/swagger.json:15257]: I just did that, let's begin integration roulette once again.

**Comment by Argh4k** [comment]
Looks like an issue with open-api spec still.

**Comment by MadhavJivrajani** [comment]
If you mean with pull-kubernetes-verify, then that's passed :)

**Comment by Argh4k** [comment]
🎉

**Pull request merged**